openpype/plugins/publish/collect_subset_group.py

```diff
@@ -0,0 +1,94 @@
+"""Produces instance.data["subsetGroup"] data used during integration.
+
+Requires:
+    dict -> context["anatomyData"] *(pyblish.api.CollectorOrder + 0.49)
+
+Provides:
+    instance -> subsetGroup (str)
+
+"""
+import pyblish.api
+
+from openpype.lib.profiles_filtering import filter_profiles
+from openpype.lib import (
+    prepare_template_data,
+    StringTemplate,
+    TemplateUnsolved
+)
+
+
+class CollectSubsetGroup(pyblish.api.InstancePlugin):
```

Comment:
You can't collect subset groups during collection because there are instances that yet don't have right family.

Reply:
Could you identify for me which ones those are?  Because if that's the case - then whatever logic is involved in those areas should actually move to Collectors! :)

Reply:
TVPaint rendering, Nuke rendering and some standalone publisher extractors.

Reply:
Thanks, that's good information. Here's what I found quickly:

- So in TV Paint we're changing the family "renderpass" and "renderlayer" to `render` in the Extractor.
    - Which also breaks the logic that `family` is the primary family and should also be in `families`.
    - For these cases - what's holding us back from adding `renderpass` or `renderlayer` to `instance.data["families"]` at the very beginning and set `instance.data["family"] = "render"` there too?

- [Nuke in Extract Render Local](https://github.com/pypeclub/OpenPype/blob/06a8f5014023ec1c792263fc66eb82c15159f908/openpype/hosts/nuke/plugins/publish/extract_render_local.py#L94-L105)
    - Same point, why not address this in a collector? It seems like we're breaking Pyblish workflow here.

- [Standalone Publisher changing family + families in Extractor](https://github.com/pypeclub/OpenPype/blob/8432e94615a75216e7172cf46f1370fff806bf13/openpype/hosts/standalonepublisher/plugins/publish/extract_bg_for_compositing.py#L174-L183)
- [Standalone Publisher creating new instances in Extractor](https://github.com/pypeclub/OpenPype/blob/8432e94615a75216e7172cf46f1370fff806bf13/openpype/hosts/standalonepublisher/plugins/publish/extract_images_from_psd.py#L90-L130)
    - In both these cases the PSD data should have been retrieved in a Collector itself, no?

These seem like "issues" on their own and go against the nature of Pyblish and how Validators should process the data, etc.

Worst case scenario we could work around this by moving `CollectSubsetGroup` to `IntegratorOrder - 0.499`. However to me this seems we're looking at code debt that we'll need to solve anyway and cleanup for these Extractors instead.

Reply:
I think you're right in all of these. Only remaining issue I can think of is editorial publishing where asset documents may not exists at the collection time because are created during extraction. I don't know how it's handled now, but I'm 99% sure it can't be changed.

> Worst case scenario we could work around this by moving CollectSubsetGroup to IntegratorOrder - 0.499.

If it must be moved out of integrator then it's best solution for current state.

> Standalone Publisher creating new instances in Extractor

Oh my god! ...and it was me who did it?!!!